src/QsCompiler/TestTargets/Simulation/Target/Simulation.csproj

```diff
@@ -18,29 +18,13 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="libLLVM.runtime.win-x64" Version="12.0.0" PrivateAssets="All" />
-    <PackageReference Include="libLLVM.runtime.osx-x64" Version="12.0.0" PrivateAssets="All" />
-    <PackageReference Include="libLLVM.runtime.ubuntu.20.04-x64" Version="12.0.0" PrivateAssets="All" />
-    <PackageReference Include="libLLVM.runtime.ubuntu.18.04-x64" Version="12.0.0" PrivateAssets="All" />
-    <PackageReference Include="LLVMSharp.Interop" Version="12.0.0-beta1" />
-    <PackageReference Include="Microsoft.Quantum.Qir.Runtime" Version="0.20.2110171573-alpha" ExcludeAssets="all" GeneratePathProperty="true" />
-    <PackageReference Include="Microsoft.Quantum.Simulators" Version="0.20.2110171573" GeneratePathProperty="true" />
+    <PackageReference Include="Microsoft.Quantum.Qir.Runtime" Version="0.22.187631-alpha" ExcludeAssets="compile" />
+    <PackageReference Include="Microsoft.Quantum.Simulators" Version="0.22.187631" ExcludeAssets="compile" GeneratePathProperty="true" />
   </ItemGroup>
 
-  <Target Name="PrepareReferenceTests" Condition="'$(DesignTimeBuild)' != 'true'" BeforeTargets="CoreCompile">
-    <PropertyGroup>
-      <QirRuntimeLibs Condition="$([MSBuild]::IsOsPlatform('OSX'))">$(PkgMicrosoft_Quantum_Qir_Runtime)/runtimes/osx-x64/native</QirRuntimeLibs>
-      <QirRuntimeLibs Condition="$([MSBuild]::IsOsPlatform('Windows'))">$(PkgMicrosoft_Quantum_Qir_Runtime)/runtimes/win-x64/native</QirRuntimeLibs>
-      <QirRuntimeLibs Condition="$([MSBuild]::IsOsPlatform('Linux'))">$(PkgMicrosoft_Quantum_Qir_Runtime)/runtimes/linux-x64/native</QirRuntimeLibs>
-      <SimulatorRuntime Condition="$([MSBuild]::IsOsPlatform('OSX'))">$(PkgMicrosoft_Quantum_Simulators)/runtimes/osx-x64/native/libMicrosoft.Quantum.Simulator.Runtime.dylib</SimulatorRuntime>
-      <SimulatorRuntime Condition="$([MSBuild]::IsOsPlatform('Windows'))">$(PkgMicrosoft_Quantum_Simulators)/runtimes/win-x64/native/Microsoft.Quantum.Simulator.Runtime.dll</SimulatorRuntime>
-      <SimulatorRuntime Condition="$([MSBuild]::IsOsPlatform('Linux'))">$(PkgMicrosoft_Quantum_Simulators)/runtimes/linux-x64/native/libMicrosoft.Quantum.Simulator.Runtime.so</SimulatorRuntime>
-    </PropertyGroup>
-    <ItemGroup>
-      <_QirRuntimeLibFiles Include="$(QirRuntimeLibs)/**/*.*" Exclude="$(QirRuntimeLibs)/**/*.exe" />
-    </ItemGroup>
-    <Copy SourceFiles="$(SimulatorRuntime)" DestinationFolder="$(OutputPath)" SkipUnchangedFiles="true" />
-    <Copy SourceFiles="@(_QirRuntimeLibFiles)" DestinationFolder="$(OutputPath)\%(RecursiveDir)" SkipUnchangedFiles="true" />
+  <Target Name="PrepareReferenceTests" Condition="'$(DesignTimeBuild)' != 'true' And $([MSBuild]::IsOsPlatform('OSX'))" BeforeTargets="CoreCompile">
```

Comment:
Do we still need these conditionals?

Reply:
Yes, because these libraries only need to be copied on Mac, not on any other platform (took me a while to figure that out). It has to do with how the Qir.Runtime tries to load the simulator by name here: https://github.com/microsoft/qsharp-runtime/blob/8c2a77cfd1c83e2aa2e488d9be0aef45e4505170/src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp#L52-L75. Because that is native code, it does not benefit from the extra lookup logic that .NET uses. And because it goes by name only and doesn't use any relative paths, the binaries need to be in a place that the OS was going to look already. On Windows and Linux, our preloading of the binary in .NET is sufficient to cause the later native code under the pinvoke call to succeed, but on Mac it will try to find the binary again and fail if it's not in folder next to the .NET dll. This is only for test execution, not for any production code.